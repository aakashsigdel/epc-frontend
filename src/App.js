import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

import Display from './task/Display';
import Input from './task/Input';
import Loading from './task/Loading';
import getColor from './task/api';

class App extends Component {
  state = {
    color: '#000000',
    loading: false,
    text: 'Click Me!'
  };

  renderDisplay = () => {
    const { color, loading, text } = this.state;

    return loading
      ? <Loading />
      : (
        <Display
          bgColor={ color }
          onClick={ this.getBoxColor }
          inputText={ text }
        />
      );
  }

  getBoxColor = () => {
    this.setState({ loading: true });
    getColor()
      .then(color => this.setState({ color, loading: false }));
  }

  onChangeText = e => this.setState({ text: e.target.value });

  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <Input onChangeText={this.onChangeText} />
          <h2>Welcome to React</h2>
          { this.renderDisplay() }
        </div>
      </div>
    );
  }
}

export default App;
