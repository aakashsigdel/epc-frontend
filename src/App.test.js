import React from 'react';
import App from './App';

import renderer from 'react-test-renderer';

it('should render <App /> correctly', () => {
  const component = renderer.create(<App />)
  const snapshotTree = component.toJSON();

  expect(snapshotTree).toMatchSnapshot();
});

