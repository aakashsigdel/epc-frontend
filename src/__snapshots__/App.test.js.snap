// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <App /> correctly 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <div
      className="task-container"
    >
      <input
        onChange={[Function]}
        placeholder="Type on me"
        type="text"
      />
      <div
        className="display-container"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#000000",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "grey",
              "opacity": 0.8,
            }
          }
        >
          Click Me!
        </div>
      </div>
    </div>
  </div>
</div>
`;
